docs/custom-flows/user-impersonation.mdx

```diff
@@ -102,79 +102,121 @@ The above JSON snippet shows what a Clerk session token looks like when a user i
 
 When working with frontend (client-side) code, you can easily detect impersonated sessions and gain access to the actor ID or other information from the `act` claim.
 
-<CodeBlockTabs type="framework" options={["Next.js", "React", "JavaScript"]}>
-```jsx
-import { useAuth } from "@clerk/nextjs";
-import { withServerSideAuth } from "@clerk/nextjs";
-
-// SSR authentication context
-export const getServerSideProps = withServerSideAuth(
-  async ({ req }) => {
-    // The request is augmented with an auth object.
-    // The actor object is available in there.
-    const { userId, actor } = req.auth;
+<Tabs type="framework" items={["Next.js", "React", "JavaScript"]}>
```

Comment:
Is it worth adding an app router example here?

Reply:
yesss, we have slowly but surely been adding app router examples to the docs ([DOCS-259](https://linear.app/clerk/issue/DOCS-259/app-router-code-missing-for-a-lot-of-nextjs-examples))